pydantic_ai_slim/pydantic_ai/models/anthropic.py

```diff
@@ -567,15 +623,28 @@ def _infer_tool_choice(
 
             return tool_choice
 
-    def _map_extra_headers(self, beta_features: list[str], model_settings: AnthropicModelSettings) -> dict[str, str]:
-        """Apply beta_features to extra_headers in model_settings."""
+    def _prepare_betas_and_headers(
+        self, betas: set[str], model_settings: AnthropicModelSettings
+    ) -> tuple[list[str], dict[str, str]]:
+        """Prepare beta features list and extra headers for API request.
+
+        Handles merging custom anthropic-beta header from extra_headers into betas set
```

Comment:
Always use backticks for inline code

```suggestion
        Handles merging custom `anthropic-beta` header from `extra_headers` into betas set
```

Also `User-Agent` and `extra_headers` below